avocado/utils/linux_modules.py

```diff
@@ -103,9 +103,9 @@ def loaded_module_info(module_name):
              dependent on, list of dictionary of param name and type
     :rtype: dict
     """
-    l_raw = process.system_output('/sbin/lsmod')
+    l_raw = process.run('/sbin/lsmod', verbose=True).stdout_text.rstrip('\n\r')
     modinfo_dic = parse_lsmod_for_module(l_raw, module_name)
-    output = process.system_output("/sbin/modinfo %s" % module_name)
+    output = process.run("/sbin/modinfo %s" % module_name, verbose=True).stdout_text.rstrip('\n\r')
```

Comment:
Hello @chunfuwen I spent some time on the `process` library, it's quite broken in terms of encoding. For now please make it `process.system_output().decode('utf-8')` which should work well and we can get it in.

In general where `process.run` is used, use `process.run.stdout_text` and where specific call like `process.system_output` is used use the `decode`. In case the process requires `utf-8` chars force `encoding='utf-8'` when calling it until we have the `astring.ENCODING`. When that PR is merged start using `astring.ENCODING` instead of `utf-8`.

Later I'll send the `process` fixes that should minimize the need of forcing the encoding anyway I found several issues there (mainly we need to run encoded commands instead of decoded ones and the sudo handling is all broken).

Reply:
PS: For `ASCII` the process library works well and when `process.run(..., encoding='utf-8')` is set on UTF-8 machine it also works well. It's just that the whole encoding handling is broken and only works when system one matches the used one. I'm working on proper handling.

Reply:
PPS: After the `process` library is fixed I'll go through avocado.core and avocado.utils to fix all places where `utf-8` is hardcoded, but I don't want to give any false hopes on when it could be delivered. So please watch out for `astring.ENCODING` and follow the advice I mentioned earlier.